src/hashing.rs

```diff
@@ -468,10 +575,50 @@ mod test {
         );
     }
 
+    #[test]
+    fn digest_is_the_same_as_standard_api() {
+        hash_domain!(MyDemoHasher, "com.macro.test");
+        util::hash_test::<DomainSeparatedHasher<Blake256, MyDemoHasher>>(
+            &[0, 0, 0],
+            "5faa7d48b551362bbee8a02c43e6ab634ed47c58ecf7b353f9afedfe3d574608",
+        );
+
+        let mut hasher = DomainSeparatedHasher::<Blake256, MyDemoHasher>::new("");
+        hasher.update(&[0, 0, 0]);
+        let hash = hasher.finalize();
+        assert_eq!(
+            to_hex(hash.as_ref()),
+            "5faa7d48b551362bbee8a02c43e6ab634ed47c58ecf7b353f9afedfe3d574608"
+        );
```

Comment:
I think it is important to show that this equality only holds if an empty label is supplied for the same hash domain, thus add a failing test for a hasher with a label.

Otherwise LGTM